storage/innobase/srv/srv0start.cc

```diff
@@ -3913,10 +3954,30 @@ static lsn_t srv_shutdown_log() {
 
   srv_wake_log_tracker_thread();
 
-  if (srv_downgrade_logs) {
+  /* if upgrade is failed, delete and create new empty redo so old version do
+   not complaint  */
```

Comment:
If the upgrade fails, new redo log format is not compatible with older versions, recreate them

Reply:
fixed